src/targets/gpu/kernels/include/migraphx/kernels/hip.hpp

```diff
@@ -24,11 +24,17 @@
 #ifndef MIGRAPHX_GUARD_KERNELS_HIP_HPP
 #define MIGRAPHX_GUARD_KERNELS_HIP_HPP
 
+#ifndef MIGRAPHX_USE_HIPRTC
 // Workaround macro redefinition issue with clang tidy
 #if defined(__HIP_PLATFORM_HCC__) && defined(MIGRAPHX_USE_CLANG_TIDY)
 #undef __HIP_PLATFORM_HCC__ // NOLINT
 #endif
-
 #include <hip/hip_runtime.h>
+#include <hip/hip_fp16.h>
+#include <hip/math_functions.h>
+#else
+#include <hip/hip_common.h>
```

Comment:
We shouldn't include headers from hip when using hiprtc. If we need the functionality from those headers then we should open an issue about this.

For this PR, just include it inside `#ifdef MIGRAPHX_ENABLE_HIPRTC_WORKAROUNDS` and then add that define when the env variable is set.

Reply:
`hip_common.h` is only so that we get defines for `HIP_PLATFORM_AMD` which is required for `hip_math_constants.h`. 

`hiprtc.h` includes `hip_common.h` [internally](https://github.com/ROCm-Developer-Tools/HIP/blob/436ba9757b965ef967e74f2a9688be5ee0ea863b/include/hip/hiprtc.h#L24).

Reply:
> hip_common.h is only so that we get defines for HIP_PLATFORM_AMD which is required for hip_math_constants.h.

Yes but hiprtc should provide those defines for us without including the file, as those file may not be available during runtime.

> hiprtc.h includes hip_common.h [internally](https://github.com/ROCm-Developer-Tools/HIP/blob/436ba9757b965ef967e74f2a9688be5ee0ea863b/include/hip/hiprtc.h#L24).

Yea but thats on the host side, we need this for device compilation.

Reply:
> Yes but hiprtc should provide those defines for us without including the file, as those files may not be available during runtime. 

Aren't both HIPRTC and hip-clang are shipping under the same library `libamdhip64.so`.  If so, those headers files would be available.

Reply:
> Aren't both HIPRTC and hip-clang are shipping under the same library libamdhip64.so.

Thats not a header thats a .so. The .so is included in the runtime package, but the headers need to be installed using the dev package. We cant require dev packages when using migraphx's runtime package.

Reply:
okay, 
I see also see comment in hipRTC header that, eventually it will be shipped as separate library. 

I only need hip_common.h and hip_math_constants.h for `PI_2`.  we can also just define that our side for better workaround.